daffodil-lib/src/test/scala/org/apache/daffodil/lib/util/TestFixedICUDecimalFormat.java

```diff
@@ -0,0 +1,159 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.daffodil.lib.util;
+
+import com.ibm.icu.text.DecimalFormat;
+import com.ibm.icu.text.DecimalFormatSymbols;
+import org.junit.Test;
+import static junit.framework.TestCase.assertEquals;
+import static org.junit.Assert.*;
+
+import java.util.Locale;
+import java.util.Objects;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+/**
+ * Tests about ICU-22558
+ *
+ * https://unicode-org.atlassian.net/browse/ICU-22558
+ */
+public class TestFixedICUDecimalFormat {
+
+    // Create DecimalFormatSymbols for a Locale where the decimal separator is a dot
+    DecimalFormatSymbols symbols = new DecimalFormatSymbols(Locale.US);
+
+    /**
+     * Illustrates the bug in ICU.
+     */
+    @Test
+    public void testDecimalFormatWithHash() {
+        // Create DecimalFormat instance with the pattern "#.##"
+        DecimalFormat decimalFormat = new DecimalFormat("#.##", symbols);
```

Comment:
Seems like a patterm of `.##` gives the expected results

```scala
scala> new com.ibm.icu.text.DecimalFormat(".##").format(0.12)
res1: String = .12
```

Is that a reasonable workaround?

Reply:
No, because 1.0 then fails with a truncation no-rounding possible error.

Making the user use two different format patterns is also no good as that would require the schema author to understand this bug, and literally conditionalize their schema for whether the value is between 0.0 and 1.0.

Reply:
This works for me:
```scala
scala> new com.ibm.icu.text.DecimalFormat(".##").format(1.0)
res0: String = 1.0
```

I guess we're setting some other property that is different than what a default DecimalFormat gets to cause the no-rounding error?

Reply:
I wrote the nested loop:
```
    @Test
    public void icuTesting() {
        List<String> stringList = Arrays.asList(".", "#.", ".#", "#.#");
        List<Double> numberList = Arrays.asList(0.0, 1.0, 0.1, 1.1);
        for (String str : stringList) {
            for (Double num : numberList) {
                DecimalFormat df = new DecimalFormat(str);
                String output = df.format(num);
                System.out.println(String.format("For value %f and pattern '%s' output is '%s' (length %d)", num, str, output, output.length()));
            }
        }
    }
```

Output is, plus my comments after the //. 
```
For value 0.000000 and pattern '.' output is '0.' (length 2)  // so leaving off the # entirely means the fraction digit is optional, but not the decimal point. 
For value 1.000000 and pattern '.' output is '1.' (length 2)
For value 0.100000 and pattern '.' output is '0.' (length 2)
For value 1.100000 and pattern '.' output is '1.' (length 2)   // So "." behaves exactly like "#."

For value 0.000000 and pattern '#.' output is '0.' (length 2) // leaving off the # after "." =>  insisting on one in the output???
For value 1.000000 and pattern '#.' output is '1.' (length 2)  
For value 0.100000 and pattern '#.' output is '0.' (length 2)
For value 1.100000 and pattern '#.' output is '1.' (length 2)

For value 0.000000 and pattern '.#' output is '.0' (length 2) // notice not a leading integer digit here
For value 1.000000 and pattern '.#' output is '1.0' (length 3)
For value 0.100000 and pattern '.#' output is '.1' (length 2)
For value 1.100000 and pattern '.#' output is '1.1' (length 3)

For value 0.000000 and pattern '#.#' output is '0' (length 1) // suppresses trailing "."
For value 1.000000 and pattern '#.#' output is '1' (length 1) // suppresses trailing "."
For value 0.100000 and pattern '#.#' output is '0.1' (length 3)
For value 1.100000 and pattern '#.#' output is '1.1' (length 3)
```
I can come up with no rationale for this other than they have to preserve backward compatibility so this clearly ad-hoc behavior must stay.